src/v1/unpack.rs

```diff
@@ -244,43 +244,30 @@ impl<'a> PackStreamDecoder<'a> {
         Ok(byte)
     }
 
-    fn read_2_bytes(&mut self) -> PyResult<[u8; 2]> {
-        Ok([self.read_byte()?, self.read_byte()?])
-    }
-
-    fn read_4_bytes(&mut self) -> PyResult<[u8; 4]> {
-        Ok([
-            self.read_byte()?,
-            self.read_byte()?,
-            self.read_byte()?,
-            self.read_byte()?,
-        ])
-    }
-
-    fn read_8_bytes(&mut self) -> PyResult<[u8; 8]> {
-        Ok([
-            self.read_byte()?,
-            self.read_byte()?,
-            self.read_byte()?,
-            self.read_byte()?,
-            self.read_byte()?,
-            self.read_byte()?,
-            self.read_byte()?,
-            self.read_byte()?,
-        ])
+    fn read_n_bytes<const N: usize>(&mut self, n: usize) -> PyResult<[u8; N]> {
```

Comment:
I'm unsure if this makes a difference, but `read_byte` only advances the index if the bytes were read successfully. I think at the very least it should be consistent (slightly leaning towards what `read_byte` does).

Reply:
Presumably, if we fail to read the n bytes, there will be a panic, which will invalidate the result much further up the stack, but I guess for debugging, this makes sense.

Reply:
We could also fail without a panic by hitting the `None` case https://github.com/neo4j-drivers/neo4j-python-driver-rust-ext/pull/7/files#diff-211d7a1726bedd061faf9710e5130b44be014bd22ed67dc18be006a51021db43R255

I'm not sure we ever report the index back to Python in that case, but yeah, for debugging it might be easier to have it consistent.

Reply:
That's a great point!